examples/bri-3/src/bri/workgroup/worksteps/api/worksteps.controller.spec.ts

```diff
@@ -58,18 +57,19 @@ describe('WorkstepController', () => {
       ],
     })
       .overrideProvider(WorkstepStorageAgent)
-      .useValue(new MockWorkstepStorageAgent(mapper))
+      .useValue(mockDeep<WorkstepStorageAgent>())
       .compile();
 
     wController = app.get<WorkstepController>(WorkstepController);
-
+    workstepStorageAgentMock = app.get(WorkstepStorageAgent);
     await app.init();
   });
 
   describe('getWorkstepById', () => {
     it('should throw NotFound if non existent id passed', () => {
       // Arrange
       const nonExistentId = '123';
+      workstepStorageAgentMock.getWorkstepById.mockResolvedValueOnce(undefined);
```

Comment:
we are throwing an exception for other cases, why is this different?

Reply:
not sure, it seems on some places we are throwing in handler in some in agent, need issue to unify it, not with this PR though

Reply:
We are moving away from throwing exceptions on the storage agent level as they should be agnostic to the context they are called in and because of that must not stop the execution flow with an exception (i.e. if they are called from a background service such as nats messaging listener). Whenever you see a storage agent throwing exception just move the decision and the exception throw to the caller of the storage agent.